### DISCUSSION TITLE
converting ps file messes up orientation and crops content, low quality

### DISCUSSION THREAD

**Discussion by petaflot** [ImageMagick/ImageMagick#3831] [Help]
I tried to convert a ps file (displays properly with gv) to PNG and PDF, and the orientation of the page format is changed from landscape to portrait and the content is cropped.

also, the quality of the output is unsatisfying and using -size seems to have no effect

original file:
http://lutolf.vserver.nimag.net/files/primes/ftimes_4.6e4.ps

gv displays correctly:
http://lutolf.vserver.nimag.net/files/primes/ftimes_4.6e4.ps_scrn.png

bad PNG and PDF output:
http://lutolf.vserver.nimag.net/files/primes/ftimes_4.6e4.ps.png
http://lutolf.vserver.nimag.net/files/primes/ftimes_4.6e4.ps.pdf

**Comment by snibgo** [comment]
What versions of IM and Ghostscript? On what platform?

For me, IM 7.0.8-64 and GS v9.19, I get the blue at the top, as required.

Your ps file is vector. When rasterizing, the detail you get depends on the density, pixels per inch. This seems reasonable:
```
magick -density 300 ftimes_4.6e4.ps x.png
```

**Comment by petaflot** [comment]
> What versions of IM and Ghostscript? On what platform?


plaftorm: Gentoo

ebuilds:

media-gfx/imagemagick-7.0.11.12
app-text/ghostscript-gpl-9.53.3-r5



> For me, IM 7.0.8-64 and GS v9.19, I get the blue at the top, as required.


I get the blue at the top too, but the output is portrait and _not_  
landscape (and the right side of the image is cropped)


> Your ps file is vector. When rasterizing, the detail you get depends  
> on the density, pixels per inch. This seems reasonable:
> ```
> magick -density 300 ftimes_4.6e4.ps x.png



I got decent quality using


$ ps2pdf -r2400 infile.ps outfile.pdf


since this is for printing I don't really care about PNG (this was  
just so I could have fast previews for files that are close to 1GB)
-- 
The e-mail address used as "sender" is a temporary e-mail address. If  
you feel you need to keep contact with me for more than a few weeks or  
months, please request a permanent, personal e-mail address to which  
to send your messages.

**Comment by snibgo** [comment]
ImageMagick "-verbose" will tell you what dimensions IM will tell Ghostscript to make the image. I can't see where in the PS file IM gets those dimensions from, not what alternatives might be used. Sorry.
